db/schema.sql

```diff
@@ -0,0 +1,92 @@
+);
+CREATE INDEX client_reports_task_and_time_index ON client_reports(task_id, nonce_time);
+
+-- Specifies the possible state of an aggregation job.
+CREATE TYPE AGGREGATION_JOB_STATE AS ENUM(
+    'IN_PROGRESS', -- at least one included preparation is in a non-terminal (WAITING) state
+    'FINISHED'     -- all client reports have reached a terminal state (FINISHED, FAILED, INVALID)
+);
+
+-- An aggregation job, representing the aggregation of a number of client reports.
+CREATE TABLE aggregation_jobs(
+    id                 BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, -- artificial ID, internal-only
+    task_id            BYTEA NOT NULL,                  -- task
+    aggregation_param  BYTEA NOT NULL,                  -- encoded aggregation parameter (opaque VDAF message)
+    state              AGGREGATION_JOB_STATE NOT NULL,  -- current state of the aggregation job
+
+    CONSTRAINT fk_task_id FOREIGN KEY(task_id) REFERENCES tasks(id)
+);
+
+-- An aggregation attempt for a single client report. An aggregation job logically contains a number
+-- of report aggregations. A single client report might be aggregated in multiple aggregation jobs &
+-- therefore have multiple associated report aggregations.
+CREATE TABLE report_aggregations(
+    id                  BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, -- artificial ID, internal-only
+    aggregation_job_id  BIGINT NOT NULL,  -- the aggregation job ID this report aggregation is associated with
+    client_report_id    BIGINT NOT NULL,  -- the client report ID this report aggregation is associated with
+    ord                 BIGINT NOT NULL,  -- a value used to specify the ordering of client reports in the aggregation job
+    transition          BYTEA NOT NULL,   -- encoded Transition message, representing the current preparation state of this report in this aggregation job
+
+    CONSTRAINT unique_ord UNIQUE(aggregation_job_id, ord),
+    CONSTRAINT fk_aggregation_job_id FOREIGN KEY(aggregation_job_id) REFERENCES aggregation_jobs(id),
+    CONSTRAINT fk_client_report_id FOREIGN KEY(client_report_id) REFERENCES client_reports(id)
+);
+CREATE INDEX report_aggregations_aggregation_job_id_index ON report_aggregations(aggregation_job_id);
+
+-- Information on incremental aggregation, for VDAFs that support incremental aggregation (eg prio3).
+CREATE TABLE batch_window_aggregations(
+    task_id             BYTEA NOT NULL,      -- the task ID
+    batch_window_start  TIMESTAMP NOT NULL,  -- the start of the batch window
+    aggregate_share     BYTEA NOT NULL,      -- the (incremental) aggregate share
+    report_count        BIGINT NOT NULL,     -- the (incremental) client report count
+    checksum            BYTEA NOT NULL,      -- the (incremental) checksum
+
+    PRIMARY KEY(task_id, batch_window_start)
+
+    -- TODO(brandon): decide how to count collection attempts: in this table? via collect_jobs inspection? somehow else?
+);
+
```

Comment:
We will also need to store `AggregateShareResp.encrypted_aggregate_share` from each helper. This should probably go in a new table.

Reply:
Ah, you're correct of course. (I was thinking the leader would store these in memory while constructing a result, but that doesn't work now that Collect operations are async.)